### DISCUSSION TITLE
Serialize Camera Parameters

### DISCUSSION THREAD

**Discussion by avaxman** [libigl/libigl#1961] [Q&A]
Is there a way to get/set camera parameters? for instance for the purpose of screenshooting several results on the same mesh in the exact same angle, without needing to explicitly code them all to appear in the same running instance.

**Comment by alecjacobson** [comment]
oof. I think so. serialization was one of those things in libigl that started simple, grew into a fancy system and then seemed to bit-rot immediately. I'm not sure what the current state of `igl::serialize` is (I personally never used it). However, it does seem that `ViewerCore` has hooks into it. So that seems to be the intended place to look.

Meanwhile, I get ptsd every time the viewer camera is mentioned. It's a real mess of parameters. It's definitely possible to pick out the zoom, translational shift and rotation and just save/load those. I'd have to dig a bit to see which variables those would be.
